pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1189,6 +1189,11 @@ class Timedelta(_Timedelta):
                 "Value must be Timedelta, string, integer, "
                 "float, timedelta or convertible")
 
+        if unit in {'Y', 'y', 'M'}:
+            warnings.warn("M and Y units are deprecated and "
+                          "will be removed in a future version.",
```

Comment:
you likely need to specify ``stacklevel=a_number`` here (and everywhere you use. This should be high enough that it shows up when you run the tests.

Reply:
try 3

Reply:
I *suppose* this stacklevel needs to be 2

Reply:
this worrying about the exact stacklevel is not moving things along - we don’t care about this for the most part anywhere else

Reply:
This is a very simple case, that can simply be set correctly (for Timedelta itself, not for `to_timedelta` which also might end up calling `Timedelta`)